Set up CI

progress is in `logistics/travis`.

Currently stuck with the following error:
```
* Where:
Script '/home/travis/build/flutter/packages/flutter_tools/gradle/flutter.gradle' line: 653
* What went wrong:
A problem occurred configuring project ':cloud_firestore'.
> Could not get unknown property 'android' for project ':cloud_firestore' of type org.gradle.api.Project.
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 2m 45s
```

probably need to open an issue on flutter about it/ask in the discord

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -17,11 +17,12 @@ jobs:
         flutter-version: '1.9.1+hotfix.5'
     - run: flutter pub get
     - run: flutter test
+    - run: echo ${{ secrets.store_file }} | base64 --decode > ~/keystore.jks
     - run: flutter build apk --debug --split-per-abi
       env:
         KEY_ALIAS: ${{ secrets.key_alias }}
         KEY_PASSWORD: ${{ secrets.key_password }}
-        STORE_FILE: ${{ secrets.store_file }} | base64 --decode
+        STORE_FILE: ~/keystore.jks
         STORE_PASSWORD: ${{ secrets.store_password }}
     - name: Create a Release APK
       uses: ncipollo/release-action@v1

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -17,7 +17,7 @@ jobs:
         flutter-version: '1.9.1+hotfix.5'
     - run: flutter pub get
     - run: flutter test
-    - run: echo ${{ secrets.store_file }} | base64 --decode > ~/keystore.jks
+    - run: ${{ secrets.store_file }} | base64 --decode > ~/keystore.jks
     - run: flutter build apk --debug --split-per-abi
       env:
         KEY_ALIAS: ${{ secrets.key_alias }}

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -25,8 +25,8 @@ jobs:
         KEY_PASSWORD: ${{ secrets.key_password }}
         STORE_FILE: ~/keystore.jks
         STORE_PASSWORD: ${{ secrets.store_password }}
-    - name: Create a Release APK
-      uses: ncipollo/release-action@v1
-      with:
-        artifacts: "build/app/outputs/apk/debug/*.apk"
-        token: ${{ secrets.TOKEN }}
+    # - name: Create a Release APK
+    #   uses: ncipollo/release-action@v1
+    #   with:
+    #     artifacts: "build/app/outputs/apk/debug/*.apk"
+    #     token: ${{ secrets.TOKEN }}

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -17,7 +17,7 @@ jobs:
         flutter-version: '1.9.1+hotfix.5'
     - run: flutter pub get
     - run: flutter test
-    - run: ${{ secrets.store_file }} | base64 --decode > ~/keystore.jks
+    - run: echo "${{ secrets.store_file }}" | base64 --decode > ~/keystore.jks
     - run: flutter build apk --debug --split-per-abi
       env:
         KEY_ALIAS: ${{ secrets.key_alias }}

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -19,10 +19,10 @@ jobs:
     - run: flutter test
     - run: flutter build apk --debug --split-per-abi
       env:
-        KEY_ALIAS: ${{ secret.key_alias }}
-        KEY_PASSWORD: ${{ secret.key_password }}
-        STORE_FILE: ${{ secret.store_file }}
-        STORE_PASSWORD: ${{ secret.store_password }}
+        KEY_ALIAS: ${{ secrets.key_alias }}
+        KEY_PASSWORD: ${{ secrets.key_password }}
+        STORE_FILE: ${{ secrets.store_file }} | base64 --decode
+        STORE_PASSWORD: ${{ secrets.store_password }}
     - name: Create a Release APK
       uses: ncipollo/release-action@v1
       with:

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -18,6 +18,11 @@ jobs:
     - run: flutter pub get
     - run: flutter test
     - run: flutter build apk --debug --split-per-abi
+      env:
+        KEY_ALIAS: ${{ secret.key_alias }}
+        KEY_PASSWORD: ${{ secret.key_password }}
+        STORE_FILE: ${{ secret.store_file }}
+        STORE_PASSWORD: ${{ secret.store_password }}
     - name: Create a Release APK
       uses: ncipollo/release-action@v1
       with:

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -12,7 +12,7 @@ jobs:
         java-version: '12.x'
     - uses: subosito/flutter-action@v1
       with:
-        flutter-version: '1.7.8+hotfix.4'
+        flutter-version: '1.9.1+hotfix.5'
     - run: flutter pub get
     - run: flutter test
     - run: flutter build apk --debug --split-per-abi

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -1,0 +1,23 @@
+# https://medium.com/better-programming/ci-cd-for-flutter-apps-using-github-actions-b833f8f7aac
+on: push
+name: Test, Build and Release apk
+jobs:
+  build:
+    name: Build APK
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-java@v1
+      with:
+        java-version: '12.x'
+    - uses: subosito/flutter-action@v1
+      with:
+        flutter-version: '1.7.8+hotfix.4'
+    - run: flutter pub get
+    - run: flutter test
+    - run: flutter build apk --debug --split-per-abi
+    - name: Create a Release APK
+      uses: ncipollo/release-action@v1
+      with:
+        artifacts: "build/app/outputs/apk/debug/*.apk"
+        token: ${{ secrets.TOKEN }}

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -7,6 +7,8 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v1
+      with: 
+        submodules: true
     - uses: actions/setup-java@v1
       with:
         java-version: '12.x'

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -18,6 +18,7 @@ jobs:
     - run: flutter pub get
     - run: flutter test
     - run: echo "${{ secrets.store_file }}" | base64 --decode > ~/keystore.jks
+    - run: echo "${{ secrets.google_services }}" | base64 --decode > ./android/app/google-services.json
     - run: flutter build apk --debug --split-per-abi
       env:
         KEY_ALIAS: ${{ secrets.key_alias }}

--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -27,7 +27,27 @@ apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 def keyPropertiesFile = rootProject.file("key.properties")
 def keystoreProperties = new Properties()
-keystoreProperties.load(new FileInputStream(keyPropertiesFile))
+if (keyPropertiesFile.exists()) {
+    keyPropertiesFile.withReader('UTF-8') { reader ->
+        keystoreProperties.load(reader)
+    }
+}
+def key_alias = keystoreProperties.getProperty('keyAlias')
+if (key_alias == null) {
+    key_alias = System.getenv("KEY_ALIAS")
+}
+def key_password = keystoreProperties.getProperty('keyPassword')
+if (key_password == null) {
+    key_password = System.getenv("KEY_PASSWORD")
+}
+def store_file = file(keystoreProperties.getProperty('storeFile')) // wrap this in file()
+if (store_file == null) {
+    store_file = System.getenv("STORE_FILE")
+}
+def store_password = keystoreProperties.getProperty('storePassword')
+if (store_password == null) {
+    store_password = System.getenv("STORE_PASSWORD")
+}
 
 android {
     compileSdkVersion 28
@@ -52,10 +72,10 @@ android {
 
     signingConfigs {
         release {
-	    keyAlias keystoreProperties['keyAlias']
-	    keyPassword keystoreProperties['keyPassword']
-	    storeFile file(keystoreProperties['storeFile'])
-	    storePassword keystoreProperties['storePassword']
+            keyAlias key_alias
+            keyPassword key_password
+            storeFile store_file
+            storePassword store_password
         }
     }
 

--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -43,7 +43,7 @@ if (key_password == null) {
 def store_file_path = keystoreProperties.getProperty('storeFile') // wrap this in file()
 def store_file = ""
 if (store_file_path == null) {
-    store_file = System.getenv("STORE_FILE")
+    store_file = file(System.getenv("STORE_FILE"))
 } else {
     store_file = file(store_file_path)
 }

--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -40,9 +40,12 @@ def key_password = keystoreProperties.getProperty('keyPassword')
 if (key_password == null) {
     key_password = System.getenv("KEY_PASSWORD")
 }
-def store_file = file(keystoreProperties.getProperty('storeFile')) // wrap this in file()
-if (store_file == null) {
+def store_file_path = keystoreProperties.getProperty('storeFile') // wrap this in file()
+def store_file = ""
+if (store_file_path == null) {
     store_file = System.getenv("STORE_FILE")
+} else {
+    store_file = file(store_file_path)
 }
 def store_password = keystoreProperties.getProperty('storePassword')
 if (store_password == null) {

--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -101,9 +101,9 @@ flutter {
 
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
-    testImplementation 'junit:junit:4.12'
-    androidTestImplementation 'com.android.support.test:runner:1.0.2'
-    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
+    // testImplementation 'junit:junit:4.12'
+    // androidTestImplementation 'com.android.support.test:runner:1.0.2'
+    // androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
 }
 
 apply plugin: 'com.google.gms.google-services'  // Google Play services Gradle plugin

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -1,12 +1,12 @@
 buildscript {
-    ext.kotlin_version = '1.2.71'
+    ext.kotlin_version = '1.3.10' // 1.2.71
     repositories {
         google()
         jcenter()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.2.1'
+        classpath 'com.android.tools.build:gradle:3.4.2' // 3.2.1
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
         classpath 'com.google.gms:google-services:4.3.0'
     }

--- a/android/gradle/wrapper/gradle-wrapper.properties
+++ b/android/gradle/wrapper/gradle-wrapper.properties
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.1.1-all.zip

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -36,6 +36,7 @@ dependencies:
 dev_dependencies:
   flutter_test:
     sdk: flutter
+  test: ^1.6.3
 
 
 # For information on the generic Dart part of this file, see the

--- a/test/dummy_test.dart
+++ b/test/dummy_test.dart
@@ -1,0 +1,8 @@
+// Import the test package and Counter class
+import 'package:test/test.dart';
+
+void main() {
+  test('Example', () {
+    expect(1, 1);
+  });
+}

